src/helper.py

```diff
@@ -48,9 +48,10 @@ def save_update(self, taskId, status, fileName, progress=None, fullPath=None, di
         if fullPath is not None:
             external_physical_path = f'{self.__config["fs"]["external_physical_path"]}/{directoryName}/{fileName}.{self.__config["gdal"]["output_format"]}'
             actual_size = self._convert_and_round_filesize(file_size)
-            if self.__config["storage_provider"] == StorageProvider.S3.value:
+            storage_provider = self.__config["storage_provider"].upper()
```

Comment:
storage_provider = self.__config["storage_provider"].upper()

why twice? 
can't it be done once?